### PR TITLE
contracts-bedrock: fully deprecate semver

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Removes the implementation of `Semver` in favor of `ISemver` fully.
Now all contracts that previously used `Semver` should use `ISemver`
and manually implement the `version()(string)` getter. Also update
some import paths in files that were modified with the `Semver` change.

Also removes any semver from `SchemaResolver` because it is an abstract
contract. The contracts that inherit it should implement `ISemver` so
that they can be versioned independently.

Each contract that had its source code updated had a minor version bump.
Regenerate the bindings, gas-snapshot and semver-lock.

Note that the `OptimismMintableERC20Factory` had its semver bumped
because it deploys `OptimismMintableERC20` tokens, which had its
bytecode changed by moving from `Semver` to `ISemver`.

### CODE REVIEW

**Comment by tynes** [comment]
Getting this error in ci:

```
Error response from daemon: login attempt to https://registry-1.docker.io/v2/ failed with status: 401 Unauthorized
```

File: packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol [hunk 1]
```diff
@@ -38,6 +38,10 @@ contract OptimismMintableERC20 is IOptimismMintableERC20, ILegacyMintableERC20,
         _;
     }
 
+    /// @notice Semantic version.
+    /// @custom:semver 1.3.0
+    string public constant version = "1.3.0";
+
     /// @custom:semver 1.2.1
```
Comment (maurelian) [inline packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol]: Needs updating or deleting

File: packages/contracts-bedrock/src/periphery/op-nft/OptimistInviter.sol [hunk 2]
```diff
@@ -87,10 +87,14 @@ contract OptimistInviter is Semver, EIP712Upgradeable {
     /// @notice Maps from addresses to number of invites they have.
     mapping(address => uint256) public inviteCounts;
 
+    /// @notice Semantic version.
+    /// @custom:semver 1.1.0
+    string public constant version = "1.1.0";
+
     /// @custom:semver 1.0.2
```
Comment (maurelian) [inline packages/contracts-bedrock/src/periphery/op-nft/OptimistInviter.sol]: Delete

File: packages/contracts-bedrock/src/periphery/op-nft/OptimistAllowlist.sol [hunk 3]
```diff
@@ -30,6 +30,10 @@ contract OptimistAllowlist is Semver {
     ///         attestations.
     address public immutable OPTIMIST_INVITER;
 
+    /// @notice Semantic version.
+    /// @custom:semver 1.1.0
+    string public constant version = "1.1.0";
+
     /// @custom:semver 1.0.2
```
Comment (maurelian) [inline packages/contracts-bedrock/src/periphery/op-nft/OptimistAllowlist.sol]: Delete

File: packages/contracts-bedrock/test/Semver.t.sol
Comment (maurelian) [inline packages/contracts-bedrock/test/Semver.t.sol:1]: We should also have a test for ISemver

**Comment by tynes** [comment]
Good catch on those leftover comments.

Quick question - how do you test semver? There isnt a generic way to test anything about it, its just a `version()(string)` method, the compiler enforces testable things by only compiling correct things by that constrain.

**Review by maurelian — APPROVED** [review]

**Pull request merged**